go/api-frontend/aaa/token_backend.go

```diff
@@ -59,3 +73,38 @@ func (ti *TokenInfo) AdminActions() map[string]bool {
 
 	return adminRolesMap
 }
+
+func ValidTokenExpiration(ti *TokenInfo, expiration time.Time, max time.Duration) (*TokenInfo, time.Time) {
+	if time.Now().Sub(ti.CreatedAt) > max {
+		// Token has reached max expiration
+		return nil, time.Unix(0, 0)
+	}
+
+	if expiration.After(ti.CreatedAt.Add(max)) {
+		expiration = ti.CreatedAt.Add(max)
+	}
+
+	return ti, expiration
+}
+
+func MakeTokenBackend(args []string) TokenBackend {
+	switch args[0] {
```

Comment:
We should probably validate that the args length is greater than 0 to enter this switch case, otherwise that can create a panic

Reply:
Done